kernel/src/util/net/options/mod.rs

```diff
@@ -134,7 +134,7 @@ pub fn new_raw_socket_option(
     match level {
         CSocketOptionLevel::SOL_SOCKET => new_socket_option(name),
         CSocketOptionLevel::SOL_TCP => new_tcp_option(name),
-        _ => todo!(),
+        _ => return_errno_with_message!(Errno::ENOPROTOOPT, "unsupported option level"),
```

Comment:
Could you please add some regression tests for the three `ENOPROTOOPT` error codes you added? I want to make sure they're the correct error code that Linux responds to when the socket level/socket option is invalid.